### PR TITLE
Generic FTS Table Tooltips for descendant entities are generated improperly #2960

### PR DESCRIPTION
**Pull request by WildChildCode**
In some reason HitInfoLoaderService.loadHitInfos(entityName, entityId, searchTerm) that is used to generate tooltips in FtsFilterHelper.buildTableTooltip(), do not find hits when we pass name of parent entity. But it finds them if we pass name of entity with given entityId. 
Thus we have to provide entities names while generating tooltips for it, but not an meta class name, determined in data source (legacy GUI) or container, as it was before.

### CODE REVIEW

File: modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java [hunk 1]
```diff
@@ -1821,7 +1822,24 @@ protected void applyFts() {
 
             if ((applyTo != null) && ListComponent.class.isAssignableFrom(applyTo.getClass())) {
                 if (clientConfig.getGenericFilterFtsTableTooltipsEnabled()) {
-                    filterHelper.initTableFtsTooltips((ListComponent) applyTo, adapter.getMetaClass(), searchTerm);
+                    filterHelper.initTableFtsTooltips(
+                            (ListComponent) applyTo,
+                            (listComponent, entityId) -> {
```
Comment (andreysubbotin) [inline modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java:1827]: Why are we using bi-function here? We have a reference to ListComponent, so we are able to extract metaclasses without additional bi-function parameter from the ListComponent.

File: modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java [hunk 2]
```diff
@@ -1821,7 +1822,24 @@ protected void applyFts() {
...
+                                if (listComponent instanceof Table) {
+                                    CollectionDatasource<Entity<Object>, Object> ds = listComponent.getDatasource();
+                                    TableItems tableItems = ((Table)listComponent).getItems();
```
Comment (andreysubbotin) [inline modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java:1830]: Do not use listComponent.getDatasource. Use ListComponent#getItems() and inheritors `DatasourceDataUnit, ContainerDataUnit` for metaClass extracting.

File: modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java (hunk 1 above)
Comment (WildChildCode) [reply modules/gui/src/com/haulmont/cuba/gui/components/filter/FilterDelegateImpl.java:1827]: I thought about. But first, `FilterDelegateImpl` incapsulates work with `DataSource` or `DataLoader` in `FilterDelegate#Adapter`. And second, when a look to `WebFilterHelper` class, I admit that it designed in some another abstraction layer and hasn't to depend on underlying classes, working with Data (`DataSource` or `DataLoader`). Thus, I used bi-function to let `FilterDelegateImpl` work with `DataSource` or `DataLoader`

**Comment by WildChildCode** [comment]
I've suddenly misclicked and close PR (reopen than). Sorry =)
I've added changes due to your comments.

**Pull request merged**